### PR TITLE
Add available, frozen fields to Wallet.

### PR DESCRIPTION
**Pull request by sutra**
A proposal of differentiating the available and frozen amount in wallet.

refs #919

### CODE REVIEW

File: xchange-core/src/main/java/com/xeiam/xchange/dto/account/AccountInfo.java [hunk 1]
```diff
@@ -71,6 +72,21 @@ public BigDecimal getTradingFee() {
   }
 
   /**
+   * Returns the wallet of the specified currency.
+   *
+   * @param currency one of the {@link Currencies}.
+   * @return the wallet of the specified currency, or a zero balance wallet if no wallet with such currency.
+   */
+  public Wallet getWallet(String currency) {
```
Comment (rafalkrupinski) [inline xchange-core/src/main/java/com/xeiam/xchange/dto/account/AccountInfo.java]: You could keep the wallets as a Map, and change `getWallets()` to `{new ArrayList(wallets.values());}` instead

Comment (sutra) [inline xchange-core/src/main/java/com/xeiam/xchange/dto/account/AccountInfo.java]: To make it safer, let's do this in XChange 4.0.0, just deprecated it now.

**Comment by rafalkrupinski** [comment]
The fields are private, how much safer could it be?

**Comment by sutra** [comment]
@rafalkrupinski some exchange implementation adapted multiple wallets with same currency code, for example `com.xeiam.xchange.huobi.HuobiAdapters.adaptHuobiAccountInfo(HuobiAccountInfo)`:
```
  public static AccountInfo adaptHuobiAccountInfo(HuobiAccountInfo a) {

    Wallet cny = new Wallet(CNY, a.getAvailableCnyDisplay().add(a.getFrozenCnyDisplay()).subtract(a.getLoanCnyDisplay()), "available");
    Wallet btc = new Wallet(BTC, a.getAvailableBtcDisplay().add(a.getFrozenBtcDisplay()).subtract(a.getLoanBtcDisplay()), "available");
    Wallet ltc = new Wallet(LTC, a.getAvailableLtcDisplay().add(a.getFrozenLtcDisplay()).subtract(a.getLoanLtcDisplay()), "available");

    // loaned wallets
    Wallet cnyLoan = new Wallet(CNY, a.getLoanCnyDisplay(), "loan");
    Wallet btcLoan = new Wallet(BTC, a.getLoanBtcDisplay(), "loan");
    Wallet ltcLoan = new Wallet(LTC, a.getLoanLtcDisplay(), "loan");

    Wallet cnyWallet = adaptWallet(CNY, a.getAvailableCnyDisplay(), a.getFrozenCnyDisplay(), a.getLoanCnyDisplay());
    Wallet btcWallet = adaptWallet(BTC, a.getAvailableBtcDisplay(), a.getFrozenBtcDisplay(), a.getLoanBtcDisplay());
    Wallet ltcWallet = adaptWallet(LTC, a.getAvailableLtcDisplay(), a.getFrozenLtcDisplay(), a.getLoanLtcDisplay());

    List<Wallet> wallets = Arrays.asList(cny, btc, ltc, cnyLoan, btcLoan, ltcLoan, cnyWallet, btcWallet, ltcWallet);
    return new AccountInfo(null, wallets);
  }
```
if change to Map<String, Wallet>, then only one of them could be kept.

**Comment by rafalkrupinski** [comment]
How about using currency+"-loan" or something similar? It's exchange specific any way.

**Comment by rafalkrupinski** [comment]
What I don't like here is that loop each time I want a Wallet for a currency.
But now you've pointed out that I cannot tell which Wallet I get - "loan" or standard.

**Comment by sutra** [comment]
I am considering to use: `accountInfo.getWallet("BTC").getLoan()` to return the amount of the loan.

**Comment by rafalkrupinski** [comment]
+1 either that or just skip it and leave it to exchange specific interface. Is it commonly used?

**Comment by sutra** [comment]
I am not using the `borrowed` or the `loan`, so I haven't implemented this in this PR. Maybe someone else could vote for this `borrowed`, `loan`....

**Pull request merged**